pycbc/inference/burn_in.py

```diff
@@ -252,13 +299,27 @@ class BaseBurnInTests(object):
     available_tests = ('halfchain', 'min_iterations', 'max_posterior',
                        'posterior_step', 'nacl',
                        )
+    available_sampler_test = ('chain_moments',
+                              )
 
     # pylint: disable=unnecessary-pass
 
     def __init__(self, sampler, burn_in_test, **kwargs):
         self.sampler = sampler
         # determine the burn-in tests that are going to be done
-        self.do_tests = get_vars_from_arg(burn_in_test)
+        self.do_all_test=get_vars_from_arg(burn_in_test)
+        self.do_test = [x for x in self.do_all_test if x in available_tests]
+        self.do_sampler_test = [x for x in self.do_all_test 
+                                if x in available_sampler_tests]
+        self.unrecognised_tests = [x for x in self.do_all_test 
+                                   if x not in available_tests and 
+                                   x not in available_sampler_tests]
+        #Raise an error of the burn in test is not listed
+        if self.unrecognised_tests:
+            raise ValueError("Following burn in test(s) not recognised: ",
+                             "%s"%self.unrecognised_tests,
+                             """Choose from the following tests: """,
+                             *available_tests,*available_sampler_test)
```

Comment:
You don't need to store unrecoginized tests here, since you're just using that to check for errors. Also, it's simpler to do this using sets:
```
unrecognized_tests = (set(all_tests) - set(self.do_tests)) - set(self.do_sampler_tests)
if unrecognized_tests:
     raise ValueError("Following burn in test(s) not recognized: {}"
                      "Chose from: {}".format(', '.join(unrecognized_tests), 
                      ', '.join(list(self.available_tests)+list(self.available_sampler_tests))))
```